### PR TITLE
week 3

### PR DESCRIPTION
**Pull request by HieuNC7**

### CODE REVIEW

**Comment by HieuNC7** [comment]
create user:
image
get list user:
image
db query:
image

File: demo/src/main/java/com/springboot/demo/week3/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/user")
```
Comment (hieu-nv) [inline demo/src/main/java/com/springboot/demo/week3/controller/UserController.java:17]: Should use plural for path mapping

File: demo/src/main/java/com/springboot/demo/week3/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+public class UserController {
+
+  private final UserService userService;
+
+  @GetMapping("/get-list-user")
```
Comment (hieu-nv) [inline demo/src/main/java/com/springboot/demo/week3/controller/UserController.java:22]: For list user use can use `GET /users` path Mapping

File: demo/src/main/java/com/springboot/demo/week3/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  public ResponseEntity<ResponseListUser> getListUser() {
+    return ResponseEntity.ok(userService.getListUser());
+  }
+
+  @PostMapping("/create-user")
```
Comment (hieu-nv) [inline demo/src/main/java/com/springboot/demo/week3/controller/UserController.java:27]: For create user, you can use `POST /users`

File: demo/src/main/java/com/springboot/demo/week3/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+
+  private final UserRepository userRepository;
+
+  @Override
+  public ResponseListUser getListUser() {
```
Comment (hieu-nv) [inline demo/src/main/java/com/springboot/demo/week3/service/impl/UserServiceImpl.java:20]: DTO for response data should be suffix with `Response1

File: demo/src/main/java/com/springboot/demo/week3/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    ResponseListUser res = new ResponseListUser();
+    List<User> listUser = userRepository.findAllUser();
+    res.setData(listUser);
+    res.setTotal(listUser.size());
+    return res;
+  }
+
+  @Override
+  public ResponseCreateUser createUser(CreateUserDto userDto) {
+    ResponseCreateUser res = new ResponseCreateUser();
```
Comment (hieu-nv) [inline demo/src/main/java/com/springboot/demo/week3/service/impl/UserServiceImpl.java:30]: Should use builder to create instance

File: demo/src/main/java/com/springboot/demo/week3/controller/UserController.java (hunk 1 above)
Comment (hieu-nv) [reply demo/src/main/java/com/springboot/demo/week3/controller/UserController.java:17]: /user -> /users

`GET /users` list, array

`POST /users` -> data in request body

`PUT /users/{userId}` ->  

`DELETE /users/{userId}`

`DELETE /users`

`PATCH /users` - 1 property

**Pull request merged**